services/placement/types.go

```diff
@@ -36,13 +36,39 @@ type Algorithm interface {
 	// RemoveInstances removes a list of instances from the placement.
 	RemoveInstances(p services.Placement, leavingInstanceIDs []string) (services.Placement, error)
 
-	// ReplaceInstance replace a instance with new instances.
-	ReplaceInstance(p services.Placement, leavingInstanceID string, addingInstances []services.PlacementInstance) (services.Placement, error)
+	// ReplaceInstance replace a list of instances with new instances.
+	ReplaceInstances(
+		p services.Placement,
+		leavingInstanecIDs []string,
+		addingInstances []services.PlacementInstance,
+	) (services.Placement, error)
 
 	// IsCompatibleWith checks whether the algorithm could be applied to given placement.
 	IsCompatibleWith(p services.Placement) error
 }
 
+// InstanceFilter filters valid instances for the placement change.
+type InstanceFilter interface {
```

Comment:
As mentioned above, would call this `InstanceSelector`, with `Filter` replaced with `Selector` for the methods defined in the interface.